src/main/java/com/teragrep/cnf_01/PropertiesConfiguration.java

```diff
@@ -0,0 +1,119 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.cnf_01;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/**
+ * Provides unmodifiable configuration for Java's Properties object. Uses System properties by default if the
+ * constructor is not given a parameter.
+ */
+public final class PropertiesConfiguration implements Configuration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PropertiesConfiguration.class);
+
+    private final Map<String, String> configuration;
+
+    public PropertiesConfiguration() {
+        this(System.getProperties());
+    }
+
+    public PropertiesConfiguration(final Properties properties) {
+        this(
+                Collections
+                        .unmodifiableMap(
+                                properties
+                                        .entrySet()
+                                        .stream()
+                                        .collect(
+                                                Collectors
+                                                        .toMap(k -> k.getKey().toString(), k -> k.getValue().toString())
```

Comment:
using value `k` is not very descriptive here since it usually refers to the key value. Could you use `entry` instead?

Reply:
changed to entry